### ISSUE TITLE
Should carried fighters & drones help recharge their carrier?

### ISSUE THREAD

**Issue by tehhowch** [endless-sky/endless-sky#3174]
Currently, only the parent helps the carried ship, and if the carried ship is fully repaired, any shield or hull repair ability it has is not used, even if the parent could benefit from it.

As-is, this innate repair functionality of the carried ship is cost-free - the carried ship gets to use it no matter how much energy it has and how much energy is needed to perform the repair. (This may warrant fixing). (edit: this cost is added to the parent in #3182 )

If the shared repairs are made into a two-way street, then the costs of the carried ship's repair functionality will need to be applied to the parent, lest players load their repair outfits into fighters and then simply never deploy fighters.

**Comment by Amazinite** [comment]
This does sound like it could be an interesting mechanic to have and use, but it does beg a question of why not all functions of a ship (energy, cooling, and the mentioned hull and shield recharge) should be shared between carrier and carried. If a docked ship is fully charged, should the excess energy not go to the carrier, with the fighter and carrier sharing the heat generated? And for shield regeneration or hull repair, the generator which the fighter has equipped will likely already be sufficient to sustain these outfits, essentially giving the carried ship free repairs at maybe only the cost of heat, unless of course for some reason the fighter is able to provide repairs to the carrier but unable to provide energy as well.

To me, it makes sense that a carried ship would be able to acquire repairs from its mothership when docked but the fighter would be unable to assist the carrier. To have the carried ship provide shield regen and hull repair to the carrier also seems to me as if it would be a huge boost to the player (it is essentially increased outfit space for carrier ships) but a generally unused mechanic by the AI.

TL;DR, cool idea, I'd like it, but I question the balance and rationality of it.

**Comment by Nechochwen-D** [comment]
The only thing I can see doing is allowing power generation to be shared.

**Comment by ghoulavenger** [comment]
Considering that fighters have extremely limited outfit space, I don't see how the parent ship would benefit all that much from outfits installed on fighters. Even if you were to remove the weapon mounts from fighters that only adds a slight benefit for repair outfits or shield generators. The amount of energy produced and energy stored will not be significant enough to bother with on a carrier either. So while I understand the Amazinites objection, unless we see fighters with significantly more outfit space, I don't see a real problem. Conversely, I don't see a really large benefit either.

That being said, this idea does propose something I do find interesting. Fighters are only good if you're willing to spend exorbitant amounts of money on firepower (since fighters are easily destroyed, it might even be more costly than using secondary weapons) or use boxwings for extra cargo capacity (which is still significantly worse than using a dedicated freighter like the Deep River). This third option might actually be better than both and make carriers more interesting. I do think you'd need a new fighter type for it though (one with either innate bonuses in and of itself like pug ships, or one without gun/turret mounts specifically designed to benefit the carrier which will open up outfit space possibilities -- literally an extended outfit space for the carrier), otherwise it's likely to be too small an effect to matter. But if you did make these special fighters than Amazinite might be right, and it is too OP for carriers, since it is effectively a buff for any ship with fighter capacity.

**Comment by KiLEdEnNis** [comment]
you can basically remove every outfit on a fighter, then only put power & shield generators in, and do the 2x-Esc & load trick, so say a Republic Carrier has several hundred more outfit space from fighters & drones, basically even doubling outfit space (just in some separated carried ships)
so, indeed, fighters shouldn't be able to repair/recharge/anything its carrier, just the other way around for shield/hull repair

**Comment by Amazinite** [comment]
>Considering that fighters have extremely limited outfit space,

Fighters have anywhere from 85 to 110 outfit space. (The Boxwing has 45 stock but converting its 80 cargo space to 60 outfit space yields 105 total outfit space).
For drones, this number is anywhere from 53 to 86.

Assuming you use the fighters and drones with the highest outfit space, that's 110 extra space PER fighter bay and 86 extra space PER drone bay.

Ship  | Extra space
------|------
Tek Far 109 | 1544
Tek Far 78 - Osk | 990
Carrier | 956
Tek Far 71 - Lek | 860
Pond Strider | 688
Skein | 660
Solifuge | 660
Roost | 440
Cruiser | 344
Bactrian | 330
Korath Raider | 220
Nest | 220
Aerie | 220
Mule | 110

For the Tek Far 109, that might not be 1544 of pure outfit space (the fighters aren't going to allow it to install an extra triple plasma core), but cooling, shield generators, and hull repair are all small enough to pile on to those fighters and drones, and there are plenty of very powerful power generators small enough to fit on those fighters and drones.

**Comment by tehhowch** [comment]
If I were to implement bidirectional sharing, it would likely only be for repair functions. If the fighter was allowed to generate energy and heat and contribute those to the carrier, displaying this information is not possible in the current UI: a player would need to know about this functionality in order to outfit a carrier that performs adequately with both docked and undocked fightercraft. Any unsuspecting player would wonder why their ship overheats uncontrollably, or runs out of energy after launching, etc. It would also only make sense to share cooling for bays which are external, which doesn't help reduce the ridiculous benefit to a player's TF109.

Pull #1971 would create ai behaviors that result in docked fightercraft during combat, so bidirectional repair sharing would not be exclusively for the player (but certainly no AI fleets are configured to use this hypothetical relationship).

**Comment by Amazinite** [comment]
Honestly, I now think that this issue should be completely avoided, as unless there is some penalty to the effectiveness of the benefits that fighters and drones give to the carrier (so, only able to give 10% of their energy generation), then this is essentially a duplicate of #978.

**Comment by flaviojs** [comment]
Since carried ships (fighters and drones) can't assist others in space, they should NOT be able to assist the parent ship either (while docked).

Conceptually, the same hardware would be used to assist in space and assist while docked.

**Comment by tehhowch** [comment]
The lack of cost of using the fighter's native repair hardware (to heal the fighter) is addressed in #3182

**Comment by ghoulavenger** [comment]
@Amazinite I was under the impression that fighters were under the same restrictions for launching that other ships were, heating, cooling, power and thrust. So it isn't like you can just throw large systems cores on them -- which was my point. If you didn't require those attributes then, yes, it would literally be extra outfit space, in which case I agreed that they could be OP.

**Comment by Amazinite** [comment]
Yes, -18 outfit space to install the smallest thrusters and power that allow you to launch (which I disregarded originally because you use to be able to force-exit the outfitter and launch with no outfits on, but as of the latest compile the outfits check has been moved to depart and can not be forced through), but that's still a substantial extra amount of outfit space regardless.

**Comment by PJTheGuy** [comment]
Bit of a necro-post here, but I think this could be useful. I personally don't use many carriers, simply for the extra hassle they are to use. The only ones I use are Boxwings, simply for more cargo space. This could make fighters more viable in the long run. Maybe there could be a special value in the fighter description that defines it's repair rate?

**Comment by tehhowch** [comment]
Open issues are never necros!

Given the data @Amazinite collected above, and some of the conversations in referenced issues, I'm not looking to turn the recharge into a two-way street unless good arguments can be made for it. It would certainly be *neat*, but it wouldn't be particularly balanced or particularly useful for the AI - currently a (non-player) ship that enters attack mode always issues the deploy order, so any "builds" that are designed to benefit the carrier with the fighters will fail miserably in combat, as the "bay outfits" get pushed out the airlock (naively).

In a world where the AI (and player) are able to selectively deploy individual drones, such builds wouldn't be so ineffective ("does this bay's ship have any weapons? yes? ok, go fight! no? stay put!"), but they'd still be just as biased to benefit the player.

So, produce your balanced arguments and perhaps MZ or another contributor can be convinced to PR the changes. 👍

**Comment by endless-sky** [comment]
I can't see any way to make this balanced. It would turn into a way to add an extra dozen shield generators to certain ships.
